plugin.video.screenrant.com/resources/lib/screenrant.py

```diff
@@ -0,0 +1,231 @@
+
+        if self.list == None: return
+
+        for i in self.list: i.update({'action': 'play', 'isFolder': 'False'})
+
+        for i in self.list:
+            bookmark = dict((k,v) for k, v in i.iteritems() if not k == 'next')
+            bookmark['bookmark'] = i['url']
+            i.update({'cm': [{'title': 32501, 'query': {'action': 'addBookmark', 'url': json.dumps(bookmark)}}]})
+
+        for i in self.list: i.update({'nextlabel': 32500, 'nextaction': 'youtube'})
+
+        directory.add(self.list, content='files')
+        return self.list
+
+
+    def videos(self, url):
+        self.list = cache.get(self.item_list, 1, url)
+
+        if self.list == None: return
+
+        for i in self.list: i.update({'action': 'play', 'isFolder': 'False'})
+
+        for i in self.list:
+            bookmark = dict((k,v) for k, v in i.iteritems() if not k == 'next')
+            bookmark['bookmark'] = i['url']
+            i.update({'cm': [{'title': 32501, 'query': {'action': 'addBookmark', 'url': json.dumps(bookmark)}}]})
+
+        for i in self.list: i.update({'nextlabel': 32500, 'nextaction': 'videos'})
+
+        directory.add(self.list, content='files')
+        return self.list
+
+
+    def play(self, url):
+        url = self.resolve(url)
+        directory.resolve(url)
+
+
+    def item_list(self, url):
+        try:
+            url = urlparse.urljoin(self.base_link, url)
+
+            result = client.request(url)
+
+            items = client.parseDOM(result, 'article')[0]
+            items = client.parseDOM(items, 'li')
+        except:
+            return
+
+        try:
+            next, num = re.findall('(.+?/page/)(\d+)', url)[0]
+            num = int(num) + 1
+            if num > 10: raise Exception()
+            next = '%s%s/' % (next, num)
+            next = client.replaceHTMLCodes(next)
+            next = next.encode('utf-8')
+        except:
+            next = ''
+
+        for item in items:
+            try:
+                title = client.parseDOM(item, 'span', ret='data-alt')[0]
+                title = client.replaceHTMLCodes(title)
+                title = title.encode('utf-8')
+
+                url = client.parseDOM(item, 'a', ret='href')[0]
+                url = client.replaceHTMLCodes(url)
+                url = url.encode('utf-8')
+
+                image = client.parseDOM(item, 'img', ret='src')[0]
+                image = client.replaceHTMLCodes(image)
+                image = image.encode('utf-8')
+
+                self.list.append({'title': title, 'url': url, 'image': image, 'next': next})
+            except:
+                pass
+
+        return self.list
+
+
+    def resolve(self, url):
+        try:
+            if url.startswith('plugin://'): return url
+
+            url = urlparse.urljoin(self.base_link, url)
+
+            result = client.request(url)
+
+            netloc = ['brightcove', 'youtube', 'soundcloud', 'podcasts.screenrant', 'media.blubrry']
```

Comment:
hi phil
it's correct, they used to use this for their podcasts before switching to podcasts.screenrant and later to soundcloud

http://screenrant.com/screen-rant-underground-podcast-episode-6-moneyball-benk-133623/

http://media.blubrry.com/screenrant/podcast.screenrant.com/vod/podcast.screenrant/sr-underground-episode-6-moneyball.mp3